db/version_builder.cc

```diff
@@ -1020,27 +1043,17 @@ class VersionBuilder::Rep {
 
   void MaybeAddFile(VersionStorageInfo* vstorage, int level, FileMetaData* f) {
     const uint64_t file_number = f->fd.GetNumber();
-
     const auto& level_state = levels_[level];
+    const auto& del_files = level_state.deleted_base_files;
 
-    const auto& del_files = level_state.deleted_files;
-    const auto del_it = del_files.find(file_number);
-
-    if (del_it != del_files.end()) {
-      // f is to-be-deleted table file
-      vstorage->RemoveCurrentStats(f);
-    } else {
-      const auto& add_files = level_state.added_files;
-      const auto add_it = add_files.find(file_number);
-
-      // Note: if the file appears both in the base version and in the added
-      // list, the added FileMetaData supersedes the one in the base version.
-      if (add_it != add_files.end() && add_it->second != f) {
-        vstorage->RemoveCurrentStats(f);
-      } else {
-        vstorage->AddFile(level, f);
-      }
+    if (del_files.find(file_number) != del_files.end()) {
+      return;
```

Comment:
Do we assume this never happen?

Reply:
I don't think so, `MaybeAddFile` is essentially filtering base files through `deleted_base_files`.